src/Illuminate/Foundation/Bootstrap/LoadConfiguration.php

```diff
@@ -66,9 +66,10 @@ protected function loadConfigurationFiles(Application $app, RepositoryContract $
     protected function getConfigurationFiles(Application $app)
     {
         $files = [];
+        $path = realpath($app->configPath());
```

Comment:
Yeah, this is specifically to provide resilience when extending that class in order to change core paths, which may not have had `realpath()` called on them:

https://github.com/laravel/framework/blob/5.1/src/Illuminate/Foundation/Application.php#L312

If this is changed to something such as:

    return $this->basePath.DIRECTORY_SEPARATOR.$this->customConfigPath;

...realpath() won't have been called.

So in my PR this "extra" `realpath()` call only happens once, then is cached, and passed to the next function saving multiple calls to `config_path()`

So +1 fast call but -11+ slow calls, and it solves any issues with trailing slashes.

Sorry if I'm over-explaining...